### PR TITLE
[WIP] I/O names generator

### PR DESCRIPTION
**Pull request by shahor02**
@sawenzel I meant this kind of utility class (to be completed). Here I assume that only the file-names need to be detector-specific, i.e. the digit trees for different detectors will go to different files so the trees can be simply named ``digits`` (eventually also the ``hits``). Though we can also make the tree names detector-specific.

For the moment I've adapted only ITSMFTDigitWriter class, tested with 
````
o2-sim-digitizer-workflow --configKeyValues 'NameGen.ITSDigits=alt_its_digits.root;'
````
while the standard call ``o2-sim-digitizer-workflow `` will write to default ``its_digits.root``

### CODE REVIEW

File: Detectors/Base/include/DetectorsBase/NameGen.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+/// \file NameGen.h
+/// \brief Definition of the Names Generator class
+
+namespace o2
+{
+namespace base
+{
+// class for book-keeping of the names for output files and trees
+struct NameGen : public o2::conf::ConfigurableParamHelper<NameGen> {
+
```
Comment (sawenzel) [inline Detectors/Base/include/DetectorsBase/NameGen.h]: looks good to me overall. 

Not sure if "Gen" is better than just "Conf" (we are not really generating names). 
Also the keys in this parameter sometimes refer to branches sometimes to files. Should we make this part of the keyname?

Comment (shahor02) [reply Detectors/Base/include/DetectorsBase/NameGen.h]: Ok for Conf. Will add File to filenames, while the names for branches or trees are rather generic, so I would not detalize them.

**Comment by sawenzel** [comment]
I think we should move forward with this. One thing that may be improved on here is the ability to query the names gives a `DetID`.

**Comment by shahor02** [comment]
@sawenzel yes, it was postponed due to more urgent stuff.

**Comment by shahor02** [comment]
@sawenzel Could you check if this is enough for MC part?

**Comment by sawenzel** [comment]
This is part of #3075. I think we can close it to reduce our list.

**Pull request closed without merging**